tests/cases/compiler/moduleResolution_packageJson_notAtPackageRoot.ts

```diff
@@ -0,0 +1,16 @@
+// @noImplicitReferences: true
+// @traceResolution: true
+
+// Loads from a "fake" nested package.json, not from the one at the root.
+
+// @Filename: /node_modules/foo/bar/package.json
+{ "types": "types.d.ts" }
+
+// @Filename: /node_modules/foo/package.json
+{}
+
+// @Filename: /node_modules/foo/bar/types.d.ts
+export const x: number;
+
+// @Filename: /a.ts
+import { x } from "foo/bar";
```

Comment:
What about handling/testing ```foo/@bar```, ```@foo/bar```

Reply:
`@foo/bar` tested in `moduleResolution_packageJson_scopedPackage`. Changed this test to use `foo/@bar`.